server/tso/allocator_manager.go

```diff
@@ -63,78 +70,152 @@ type AllocatorManager struct {
 	//   2. Local TSO Allocator, servers for DC-level transactions.
 	// dc-location/global (string) -> TSO Allocator
 	allocatorGroups map[string]*allocatorGroup
-	// etcd and its client
-	etcd   *embed.Etcd
-	client *clientv3.Client
+	// for election use
+	member *member.Member
 	// tso config
 	rootPath      string
 	saveInterval  time.Duration
 	maxResetTSGap func() time.Duration
 }
 
 // NewAllocatorManager creates a new TSO Allocator Manager.
-func NewAllocatorManager(etcd *embed.Etcd, client *clientv3.Client, rootPath string, saveInterval time.Duration, maxResetTSGap func() time.Duration) *AllocatorManager {
+func NewAllocatorManager(m *member.Member, rootPath string, saveInterval time.Duration, maxResetTSGap func() time.Duration) *AllocatorManager {
 	allocatorManager := &AllocatorManager{
 		allocatorGroups: make(map[string]*allocatorGroup),
-		etcd:            etcd,
-		client:          client,
+		member:          m,
 		rootPath:        rootPath,
 		saveInterval:    saveInterval,
 		maxResetTSGap:   maxResetTSGap,
 	}
 	return allocatorManager
 }
 
-func (am *AllocatorManager) getAllocatorPath(dcLocation string) string {
-	// For backward compatibility, the global timestamp's store path will still use the old one
-	if dcLocation == GlobalDCLocation {
-		return am.rootPath
-	}
-	return path.Join(am.rootPath, dcLocation)
-}
-
 // SetUpAllocator is used to set up an allocator, which will initialize the allocator and put it into allocator daemon.
 func (am *AllocatorManager) SetUpAllocator(parentCtx context.Context, parentCancel context.CancelFunc, dcLocation string, leadership *election.Leadership) error {
+	var allocator Allocator
+	if dcLocation == GlobalDCLocation {
+		allocator = NewGlobalTSOAllocator(leadership, am.getAllocatorPath(dcLocation), am.saveInterval, am.maxResetTSGap)
```

Comment:
A little confused, The setup of `GlobalTSOAllocator` is only set up after the global leader is elected, while the `LocalTSOAllocator` is set up first and then elect the dc leader.

Reply:
Because the PD leader is also the `Global TSO Allocator` which means the setup work of it will be different from the `Local TSO Allocator`. For `Global TSO Allocator`, we will set it up when a PD server becomes the PD leader. At this point, the election is already done. But for `Local TSO Allocator`, the setup work will begin from the very first beginning, which needs us to run the election and some other things.

Reply:
got, maybe we can remove the `isInitialized` state in allocatorManager,  and add `IsInitailized()` interface in Allocator.  only the local allocator used the function`resetAllocatorGroup`  to reset `isInitialized`.

Reply:
Addressed.